src/backend/distributed/commands/citus_add_local_table_to_metadata.c

```diff
@@ -232,39 +237,32 @@ CreateCitusLocalTable(Oid relationId, bool cascadeViaForeignKeys)
 	 */
 	relation_close(relation, NoLock);
 
-	bool tableHasExternalForeignKeys = TableHasExternalForeignKeys(relationId);
-	if (tableHasExternalForeignKeys && cascadeViaForeignKeys)
+	if (NeedCascadeViaForeignKeys(relationId, cascadeViaForeignKeys))
```

Comment:
Accepting another boolean and throwing an error according to value of this boolean is a bit hard to reason about.
Even more, function name doesn't seem to reflect this fact.

Why do you think we should move the explicit ereport call down to this function ?